src/Stepper/Step.less

```diff
@@ -0,0 +1,205 @@
+    background-color: @neutral_dark_gray;
+    border: @iconBorderWidth solid @neutral_dark_gray;
+    color: transparent;
+  }
+}
+
+.Step--warning {
+  .Step--icon {
+    background-color: @alert_red;
+    border: @iconBorderWidth solid @alert_red;
+    color: transparent;
+  }
+}
+
+.Step--current {
+  .Step--icon {
+    background-color: @primary_blue;
+    border: @iconBorderWidth solid @primary_blue;
+    color: @neutral_white;
+  }
+
+  .Step--title, .Step--description, .Step--message--optional {
+    color: @neutral_dark_gray;
+  }
+}
+
+.Step--success:not(.Step--current) {
+  .Step--icon {
+    background-image: url("../../assets/images/check.svg");
+    background-size: @size_s;
+  }
+}
+
+.Step--warning:not(.Step--current) {
+  .Step--icon {
+    background-image: url("../../assets/images/exclamation.svg");
+    background-size: @size_2xs;
+  }
+}
+
+.Step:focus.Step--seekable, .Step:hover.Step--seekable {
+  .Step--icon {
+    border: @iconBorderWidth solid @neutral_black;
+    color: @neutral_dark_gray;
+  }
+
+  .Step--description, .Step--message--optional {
+    color: @neutral_dark_gray;
+  }
+}
+
+.Step:active.Step--seekable {
+  .Step--icon {
+    background-color: @primary_blue;
+    background-image: none;
+    border: @iconBorderWidth solid @primary_blue;
+    color: @neutral_white;
+  }
+
+  .Step--title {
+    color: @neutral_dark_gray;
+  }
+
+  .Step--description, .Step--message--optional {
+    color: @neutral_dark_gray;
+  }
+}
+
+.Step:focus.Step--seekable, .Step:focus.Step--seekable.Step--success,
+.Step:focus.Step--seekable.Step--warning, .Step:focus.Step--seekable.Step--current {
+  .Step--icon {
+    box-shadow: 0 0 0 @iconFocusShadowThickness fade(@primary_blue_tint_2, 50%);
+  }
+}
+
+.Step:focus.Step--seekable.Step--success:not(:active) {
+  .Step--icon {
+    background-color: @neutral_dark_gray;
+    border: @iconBorderWidth solid @neutral_dark_gray;
+    color: transparent;
+  }
+}
+
+.Step:focus.Step--seekable.Step--warning:not(:active) {
+  .Step--icon {
+    background-color: @alert_red_shade_2;
+    border: @iconBorderWidth solid @alert_red_shade_2;
+    color: transparent;
+  }
+}
+
+.Step:focus.Step--seekable.Step--current:not(:active) {
+  .Step--icon {
+    background-color: @primary_blue;
+    border: @iconBorderWidth solid @primary_blue;
+    color: @neutral_white;
+  }
+}
+
+.Step:hover.Step--seekable.Step--success:not(:active) {
+  .Step--icon {
+    background-color: @neutral_black;
+    border: @iconBorderWidth solid @neutral_black;
+    color: transparent;
+  }
+}
+
+.Step:hover.Step--seekable.Step--warning:not(:active) {
+  .Step--icon {
+    background-color: @alert_red_shade_2;
+    border: @iconBorderWidth solid @alert_red_shade_2;
+    color: transparent;
+  }
+}
+
+.Step:hover.Step--seekable.Step--current:not(:active) {
+  .Step--icon {
+    background-color: @primary_blue_shade_2;
+    border: @iconBorderWidth solid @primary_blue_shade_2;
+    color: @neutral_white;
+  }
+}
```

Comment:
the Step css looks a little complicated - are there ways we can simplify it?
a couple things to call out:
- i see a lot of `:not` selectors. in css, that's usually a sign that you could probably re-arrange your rules a bit to take advantage of the cascading nature of rule precedence
- i think i see some selectors that have been assigned either the same or very similar rules - helps to think about what's common and can be grouped into a single class selector higher up and then the rules that vary the ui can appear later and be much simpler because they only specify the rules that are different.